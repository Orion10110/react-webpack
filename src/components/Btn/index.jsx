import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

export default function Button ({children}) {
  return <button className="sn-btn" type="button">{children}</button>;
}

Button.propTypes = {
  children: PropTypes.string
};

Button.defaultProps = {
  children: null
};
