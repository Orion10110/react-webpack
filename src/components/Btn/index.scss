.sn-btn{
  border: none;
  background: #3f00ff;
  color: #fff;
  padding: 14px;
}