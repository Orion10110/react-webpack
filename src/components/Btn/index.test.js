import React from 'react';
import renderer from 'react-test-renderer';
import Btn from './index';

test('Render btn', () => {
  const component = renderer.create(
    <Btn>Content</Btn>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
