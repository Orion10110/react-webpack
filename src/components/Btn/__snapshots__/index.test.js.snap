// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render btn 1`] = `
<button
  className="sn-btn"
  type="button"
>
  Content
</button>
`;
